File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketPlayerSneaking.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;
import pize.tests.voxelgame.main.entity.Player;

import java.io.IOException;
import java.util.UUID;

public class CBPacketPlayerSneaking extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 17;
    
    public CBPacketPlayerSneaking(){
        super(PACKET_ID);
    }
    
    
    public UUID playerUUID;
    public boolean sneaking;
    
    public CBPacketPlayerSneaking(Player player){
        this();
        this.playerUUID = player.getUUID();
        this.sneaking = player.isSneaking();
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUUID(playerUUID);
        stream.writeBoolean(sneaking);
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        playerUUID = stream.readUUID();
        sneaking = stream.readBoolean();
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketPlayerSneaking.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.tests.voxelgame.main.entity.Player
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException
import java.util.*

class CBPacketPlayerSneaking() : IPacket<PacketHandler?>(PACKET_ID) {
    var playerUUID: UUID? = null
    var sneaking = false

    constructor(player: Player) : this() {
        playerUUID = player.uuid
        sneaking = player.isSneaking
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUUID(playerUUID!!)
        stream.writeBoolean(sneaking)
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        playerUUID = stream!!.readUUID()
        sneaking = stream.readBoolean()
    }

    companion object {
        const val PACKET_ID = 17
    }
}